pkg/util/provider/drain/drain.go

```diff
@@ -945,39 +984,45 @@ func (o *Options) waitForReattach(ctx context.Context, podVolumeInfo PodVolumeIn
 		}
 	}
 
-	klog.V(3).Infof("Successfully reattached volumes: %s", podVolumeInfo.persistentVolumeList)
+	klog.V(3).Infof("Successfully reattached volumes: %s", podVolumeInfo.volumes)
 	return nil
 }
 
-func (o *Options) getVolIDsFromDriver(ctx context.Context, pvNames []string) ([]string, error) {
-	pvSpecs := []*corev1.PersistentVolumeSpec{}
+func (o *Options) getVolumeIDFromDriver(ctx context.Context, pvName string) (string, error) {
+	var pvSpec *corev1.PersistentVolumeSpec
 
-	for _, pvName := range pvNames {
-		try := 0
+	try := 0
 
-		for {
-			pv, err := o.pvLister.Get(pvName)
+	for {
+		pv, err := o.pvLister.Get(pvName)
 
-			if apierrors.IsNotFound(err) {
-				break
-			} else if err != nil {
-				try++
-				if try == GetPvDetailsMaxRetries {
-					break
-				}
-				// In case of error, try again after few seconds
-				time.Sleep(GetPvDetailsRetryInterval)
-				continue
+		if apierrors.IsNotFound(err) {
+			return "", nil
+		} else if err != nil {
+			try++
+			if try == GetPvDetailsMaxRetries {
+				return "", err
 			}
-
-			// Found PV; append and exit
-			pvSpecs = append(pvSpecs, &pv.Spec)
-			break
+			// In case of error, try again after few seconds
+			time.Sleep(GetPvDetailsRetryInterval)
+			continue
 		}
+
+		// found PV
+		pvSpec = &pv.Spec
+		break
+	}
+
+	response, err := o.Driver.GetVolumeIDs(ctx, &driver.GetVolumeIDsRequest{PVSpecs: []*corev1.PersistentVolumeSpec{pvSpec}})
```

Comment:
Note: This can result in excessive calls to `Driver.GetVolumeIDs` and is highly-dependent on the provider implementation making this cheap. Other can result in rate limit errors at provider

Reply:
Yeah, this is true. I knowingly changed this, because existing implementation that I looked into (aws and openstack) didn't call an API on `GetVolumeIDs`.
I was under the impression that drivers are expected to extract the volume IDs purely from the PV spec without the help of an external API.

If this is not the case, I can restructure the code again to perform a single `GetVolumeIDs` call and use the list of IDs from that instead.

Reply:
@elankath can you comment on how the PR should handle this?

Reply:
Ok, the godoc for `GetVolumeIDs` does mention this (though we should fix the grammar). It is fine - you don't need to restructure.

Reply:
> If this is not the case, I can restructure the code again to perform a single `GetVolumeIDs` call and use the list of IDs from that instead.

While thinking about it again, this won't be possible.
The reason is, that the driver filters the list of input volumes depending on whether it recognizes the volume. However, we won't be able to correlate the input volumes and filtered output volumes, which would be required to achieve the needed filtering in the drain logic.

Reply:
> It is fine - you don't need to restructure.

Ok, great. Then, let's go for merging this PR :)